File: LeetCodeSolutions/brickwall.cpp
class Solution {
public:
int leastBricks(vector<vector<int>>& wall) {
	int rows = size(wall), maxEdges = 0;
	unordered_map<int, int> edgeFreq;
	for(auto& row : wall)
		for(int i = 0, idx = 0; i < size(row) - 1; i++)
		    idx += row[i], edgeFreq[idx]++, maxEdges = max(maxEdges, edgeFreq[idx]);
	return rows - maxEdges;
}

};

File: LeetCodeSolutions/brickwall.py
import collections


class Solution(object):
    def leastBricks(self, wall):
        """
        :type wall: List[List[int]]
        :rtype: int
        """
        widths = collections.defaultdict(int)
        result = len(wall)
        for row in wall:
            width = 0
            for i in xrange(len(row)-1):
                width += row[i]
                widths[width] += 1
                result = min(result, len(wall) - widths[width])
        return result
